rotkehlchen/exchanges/ftx.py

```diff
@@ -0,0 +1,514 @@
+from rotkehlchen.assets.utils import symbol_to_asset_or_token
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors import DeserializationError, RemoteError, UnknownAsset, UnsupportedAsset
+from rotkehlchen.exchanges.data_structures import AssetMovement, MarginPosition, Trade
+from rotkehlchen.exchanges.exchange import ExchangeInterface, ExchangeQueryBalances
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_asset_amount_force_positive,
+    deserialize_fee,
+    deserialize_price,
+    deserialize_timestamp_from_date,
+    deserialize_trade_type,
+)
+from rotkehlchen.typing import (
+    ApiKey,
+    ApiSecret,
+    AssetMovementCategory,
+    Fee,
+    Location,
+    Timestamp,
+)
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.interfaces import cache_response_timewise, protect_with_lock
+from rotkehlchen.utils.serialization import jsonloads_dict
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+INITIAL_BACKOFF_TIME = 2
+BACKOFF_LIMIT = 60
+PAGINATION_LIMIT = 100
+
+
+def trade_from_ftx(raw_trade: Dict[str, Any]) -> Optional[Trade]:
+    """Turns an FTX transaction into a rotki Trade.
+
+    Mary raise:
+        - UnknownAsset due to Asset instantiation
+        - DeserializationError due to unexpected format of dict entries
+        - KeyError due to dict entires missing an expected entry
+    """
+    # In the case of perpetuals and futures this fields can be None
+    if raw_trade.get('baseCurrency') is None:
+        return None
+    if raw_trade.get('quoteCurrency') is None:
+        return None
+
+    timestamp = deserialize_timestamp_from_date(raw_trade['time'], 'iso8601', 'FTX')
+    trade_type = deserialize_trade_type(raw_trade['side'])
+    base_asset = asset_from_ftx(raw_trade['baseCurrency'])
+    quote_asset = asset_from_ftx(raw_trade['quoteCurrency'])
+    amount = deserialize_asset_amount(raw_trade['size'])
+    rate = deserialize_price(raw_trade['price'])
+    fee_currency = asset_from_ftx(raw_trade['feeCurrency'])
+    fee = deserialize_fee(raw_trade['fee'])
+
+    return Trade(
+        timestamp=timestamp,
+        location=Location.FTX,
+        base_asset=base_asset,
+        quote_asset=quote_asset,
+        trade_type=trade_type,
+        amount=amount,
+        rate=rate,
+        fee=fee,
+        fee_currency=fee_currency,
+        link=str(raw_trade['id']),
+    )
+
+
+class FTX(ExchangeInterface):  # lgtm[py/missing-call-to-init]
+
+    def __init__(
+            self,
+            api_key: ApiKey,
+            secret: ApiSecret,
+            database: 'DBHandler',
+            msg_aggregator: MessagesAggregator,
+            initial_backoff: int = INITIAL_BACKOFF_TIME,
+            backoff_limit: int = BACKOFF_LIMIT,
```

Comment:
As I said in my last comment here since we don't really offer any customization to the user for those please remove them as constructor arguments and attributes of the exchange. Just have them as constants in the code.